Verification email

phulsechinmay opened the pull request:
Closes #27 Closes #28.
**Features:**
* Sends a verification link in an email once signed up
  * Verification link contains JWT (1 hour expiration) and redirect link
* Shows verification success / failure accordingly
* Verification success page has a "Continue" button that takes user to the redirect link
  * Verification success puts the auth JWT into the cookies
* Tests for all of this

**Considerations:**
* Once a user has already been verified, should be show an error if they try to go to the verification link again?
* Another issue could be added to switch the button text to "Continue to TD Registration" or "Continue to TD Forms" or whatever depending on the redirect params. Probably in a PR that personalizes the login / signup page according to redirect params too.
* Verification failure asks the user to request another verification link. Probably need to add logic for that. We could make them login and if not verified, show them an appropriate message and the ability resend verification link.

geooot commented:
@phulsechinmay 
First off, holy shit, great work! As far as your considerations go:

> Once a user has already been verified, should be show an error if they try to go to the verification link again?

It is probably a good idea just so we don't accidentally log someone in again if they clicked the link.

> Another issue could be added to switch the button text to "Continue to TD Registration" or "Continue to TD Forms" or whatever depending on the redirect params. Probably in a PR that personalizes the login / signup page according to redirect params too.

Yes figuring out how other TD Apps personalize the messaging will be a separate issue

> Verification failure asks the user to request another verification link. Probably need to add logic for that. We could make them login and if not verified, show them an appropriate message and the ability resend verification link.

Currently the messaging says that they have to login to see get a new verification link (which is fine). What we really need is logic on the JWT AuthGuard that redirects them to the verify page if they are not verified. 

This is the only thing that should be addressed in this PR

phulsechinmay commented:
@geooot By verify page, do you mean a custom page where they can request a new verification link? Or one that automatically verifies them?

phulsechinmay commented:
@geooot I've added the appropriate error view for if the user has already been verified. Also added the verification email resend view for JWT strategy for whenever a logged-in user or one trying to log in has not been verified.

Let me know what you think.
